lib/workers/global/limits.ts

```diff
@@ -37,3 +38,140 @@ export function isLimitReached(key: Limit): boolean {
   const { max, current } = limit;
   return max - current <= 0;
 }
+
+export type CountName = 'PullRequests' | 'HourlyPullRequests' | 'Branches';
+interface CountValue {
+  current: number;
+}
+type BranchLimitName =
+  | 'prHourlyLimit'
+  | 'branchConcurrentLimit'
+  | 'prConcurrentLimit';
+
+export const counts = new Map<CountName, CountValue>();
+
+export function setCount(key: CountName, val: number): void {
+  const count = val;
+  counts.set(key, { current: count });
+  logger.debug(`${key} count = ${count}`);
+}
+
+export function incCountValue(key: CountName, incBy = 1): void {
+  const count = counts.get(key) ?? { current: 0 };
+  counts.set(key, {
+    ...count,
+    current: count.current + incBy,
+  });
+}
+
+export function isCountReached(
+  key: Exclude<CountName, 'HourlyPullRequests'>,
+  config: BranchConfig,
+): boolean {
+  const limitKey =
+    key === 'Branches' ? 'branchConcurrentLimit' : 'prConcurrentLimit';
+
+  // calculate the limits for this branch
+  const hourlyLimit = calcLimit(config.upgrades, 'prHourlyLimit');
+  const limitValue = calcLimit(config.upgrades, limitKey);
+
+  const hourlyPrCount =
+    counts.get('HourlyPullRequests')?.current ??
+    // istanbul ignore next: should not happen
+    0;
+  const currentCount =
+    counts.get(key)?.current ??
+    // istanbul ignore next: should not happen
+    0;
+
+  // assuming hourly limits will be lesser than concurrent ones as it's only logical
+  if (hourlyLimit === 0 || hourlyPrCount < hourlyLimit) {
+    if (limitValue === 0 || currentCount < limitValue) {
+      return false;
+    }
+
+    return true;
+  }
+
+  return true;
+}
+
+export function calcLimit(
+  upgrades: BranchUpgradeConfig[],
+  limitName: BranchLimitName,
+): number {
+  logger.debug(
+    {
+      limits: upgrades.map((upg) => {
+        return { depName: upg.depName, [limitName]: upg[limitName] };
+      }),
+    },
+    `${limitName} of the upgrades present in this branch`,
+  );
+
+  let lowestLimit = Number.MAX_SAFE_INTEGER;
+  if (hasMultipleLimits(upgrades, limitName)) {
+    logger.warn(
```

Comment:
```suggestion
    logger.once.debug(
```
IMO should be debug.